packages/compiler/lib/decorators.cadl

```diff
@@ -4,40 +4,315 @@ using Cadl.Reflection;
 
 namespace Cadl;
 
+/**
+ * Typically a short, single-line description.
+ * @param summary Summary string.
+ *
+ * @example
+ * ```cadl
+ * @summary("This is a pet")
+ * model Pet {}
+ * ```
+ */
 extern dec summary(target: unknown, summary: string);
+
+/**
+ * Attach a documentation string.
+ * @param doc Documentation string
+ * @param formatArgs Record with key value pair that can be interpolated in the doc.
+ *
+ * @example
+ * ```cadl
+ * @doc("Represent a Pet available in the PetStore")
+ * model Pet {}
+ * ```
+ */
 extern dec doc(target: unknown, doc: string, formatArgs?: object);
+
+/**
+ * Mark a type as deprecated
+ * @param message Deprecation message.
+ *
+ * ```cadl
+ * @deprecated("Use ActionV2")
+ * op Action<T>(): T;
+ * ```
+ */
 extern dec deprecated(target: unknown, message: string);
 
+/**
+ * Service options.
+ */
 model ServiceOptions {
+  /**
+   * Title of the service.
+   */
   title?: string;
+  /**
+   * Version of the service.
+   */
   version?: string;
 }
+
+/**
+ * Mark a namespace as service namespace and configure service properties.
+ * @param options Optional configuration for the service.
+ *
+ * @example
+ * ```cadl
+ * @service
+ * namespace PetStore;
+ * ```
+ *
+ * @example Setting service title
+ * ```cadl
+ * @service({title: "Pet store"})
+ * namespace PetStore;
+ * ```
+ *
+ * @example Setting service version
+ * ```cadl
+ * @service({version: "1.0"})
+ * namespace PetStore;
+ * ```
+ */
 extern dec service(target: unknown, options?: ServiceOptions);
 
+/**
+ * Specify that this model is an error type.
+ *
+ * @example
+ * ```cadl
+ * @error
+ * model PetStoreError {
+ *   code: string;
+ *   message: string;
+ * }
+ * ```
+ */
 extern dec error(target: object);
 
+/**
+ * Specify the data format hint for a string type.
+ * @param format format name.
+ *
+ * @example
+ * ```cadl
+ * @format("uuid")
+ * scalar Uuid extends string;
+ * ```
+ */
 extern dec format(target: string | bytes | ModelProperty, format: string);
+
+/**
+ * Specify the the pattern for a string using regular expression syntax.
+ * @param pattern Regular expression.
+ *
+ * @example
+ * ```cadl
+ * @pattern("[a-z]+")
+ * scalar LowerAlpha extends string;
+ * ```
+ */
 extern dec pattern(target: string | bytes | ModelProperty, pattern: string);
 
+/**
+ * Specify the minimum length this string type should be.
```

Comment:
I like the pattern of saying "this X type" when X is the target of the decorator. Can we apply that uniformly?